test/integration/integration_test.go

```diff
@@ -234,6 +315,8 @@ func selectFieldFromLastOutput(commandField string) string {
 		outputField = lastCommandOutput.StdErr
 	case "exitcode":
 		outputField = strconv.Itoa(lastCommandOutput.ExitCode)
+	case "proxy log":
```

Comment:
@gbraad I started this way, but then I have realized that `^proxy log should contain` will be caught by `([^"]*) should contain$` if it is not first. Will be dependent on order or I will need to change `([^"]*)` for `(stdout|stderr|exitcode)`.

I agree with you that naming is bad in this case. But I am still wondering if it is better to create new functions with duplicate code here or if it would be possible to rename the existing functions and make them more re-usable. I would vote for the second, because:

```golang
func commandReturnShouldContain(commandField string, expected string) error {
	return compareExpectedWithActualContains(expected, selectFieldFromLastOutput(commandField))
}
```
there is really nothing "last command output" specific. And here:
```golang
func selectFieldFromLastOutput(commandField string) string {
	outputField := ""
	switch commandField {
	case "stdout":
		outputField = lastCommandOutput.StdOut
	case "stderr":
		outputField = lastCommandOutput.StdErr
	case "exitcode":
		outputField = strconv.Itoa(lastCommandOutput.ExitCode)
	case "proxy log":
		outputField = proxyLog.String()
	default:
		fmt.Errorf("Incorrect field type specified for comparison: %s", commandField)
	}
	return outputField
}
```

the code specific for command output is only in switch and these cases are on just one line. WDYT? Would like to ask @hferentschik also.